va/va_prot.h

```diff
@@ -0,0 +1,515 @@
+ * The most of DRM standards execute following deep pipeline to playback
+ * contents on client systems from streaming servers - provisioning uses
+ * provisioning servers, licence aquisition uses license servers, video
+ * bitstream delivery uses content servers and decryption/decoding, audio
+ * bitstream delivery uses content servers and decyption/playback,
+ * display/playback.
+ *
+ * The Provisioning and License aquisition implementations are Independent
+ * Hardware Vendor (IHV) specific but most IHVs use some form of Trusted
+ * Execution Environment (TEE) to prepare client platform or system for DRM
+ * content playback. The provisioning operations use provisioning servers (as
+ * instructed in DRM standard) and client system TEE. The communication between
+ * provisioning servers and client system TEE uses asymmetic cryptographic
+ * mechanism. This step provides a way to establish root-of-trust between
+ * client system and streaming servers. Once root-of-trust is established then
+ * client system requests for license aquisition for a particular streaming
+ * title. The license aquisition involves communication between licensing
+ * servers and TEE using asymmetic cryptographic mechanism. At end of this step,
+ * client system TEE has required assets to decrypt/decode. Although these
+ * communication does not direcly involve video aspect of GPU but **facilitate
+ * GPU required assets to playback premium contents**.
+ *
+ * To support DRM standard requirements in playback pipeline, OSes and HWs
+ * incorporate various methods to protect full playback pipeline. These 
+ * methods of protection could be SW based or HW based. The SW based protection 
+ * mechanism of DRMs is called SWDRM while HW based protection mechanism is 
+ * called HWDRM. There is no previous support in LibVA to support either DRM 
+ * mechanism.
+ * 
+ * For DRM capabilities, APIs inolve creation of protected session to 
+ * communicate with TEE and then using these protected sessions to process 
+ * video/audio data. The philophashy behind these API is to leverage existing 
+ * LibVA infrastructure as much as possible.
+ * 
+ * Note: TEE could be any secure HW device such as ME-FW or FPGA Secure
+ * Enclave or NPU Secure Enclave. There are 2 concepts here – TEE Type such
+ * as ME-FW or FPGA or NPU; TEE Type Client such as for AMT or HDCP or 
+ * something else etc.
+ *
+ * \section description Detailed Description
+ * The Protected content API provides a general mechanism for opening 
+ * protected session with TEE and if required then \ref priming GPU/Display.
+ * The behavior of protected session API depends on parameterization/
+ * configuration of protected session. Just for TEE tasks, protected 
+ * session is parameterized/configured as TEE Communication while for 
+ * HWDRM, protected session is parameterized/confgured as Protected 
+ * Content.
+ *
+ * TEE Communication Entrypoint
+ * With TEE Communication parameterization/confguration, client 
```

Comment:
s/confguration/configuration  (in multiple places)

Reply:
Fixed.